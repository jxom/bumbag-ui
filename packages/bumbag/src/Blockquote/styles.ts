import { cssClass } from '../styled';
import { darken, space, theme } from '../utils';

export const Blockquote = (styleProps) => cssClass`
  border-left: 4px solid ${darken(0.2, 'white')(styleProps)};
  padding: ${space(4)(styleProps)}rem;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
