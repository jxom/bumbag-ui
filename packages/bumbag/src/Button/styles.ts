import getDefaultPalette from '../theme/palette';
import { css, cssClass } from '../styled';
import { borderRadius, darken, palette, space, theme, fontSize, fontWeight } from '../utils';

const defaultPalette = getDefaultPalette({});

export const Button = (styleProps) => cssClass`
  align-items: center;
  background-color: ${palette(styleProps.palette)(styleProps)};
  border-radius: ${borderRadius('default')(styleProps)};
  box-shadow: 0 1px 3px 0px rgba(0, 0, 0, 0.05);
  color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
  fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
  cursor: pointer;
  display: inline-flex;
  font-weight: ${fontWeight('semibold')(styleProps)};
  min-height: 2.75rem;
  justify-content: center;
  padding: 0 ${space(4)(styleProps)}rem;
  text-decoration: none;
  hyphens: auto;
  transition: box-shadow 0.1s ease-in-out 0s, border 0.1s, background-color 0.1s;

  ${
    styleProps.palette === 'default' &&
    css`
      border: 1px solid ${palette('white900', { dark: 'gray600' })(styleProps)};
    `
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }

  &[disabled],
  &[aria-disabled="true"] {
    ${getDisabledProperties(styleProps)};
  }

  ${styleProps.size && getSizeProperties(styleProps)}
  ${styleProps.isLoading && getLoadingProperties(styleProps)};
  ${styleProps.isStatic && getStaticProperties(styleProps)};
  ${isInteractive(styleProps) && getInteractiveProperties(styleProps)};

  ${styleProps.variant === 'outlined' && getOutlinedProperties(styleProps)};
  ${styleProps.variant === 'link' && getLinkProperties(styleProps)};
  ${styleProps.variant === 'ghost' && getGhostProperties(styleProps)};
`;

export const ButtonIcon = (styleProps) => cssClass`
  ${
    styleProps.isBefore &&
    css`
      margin-left: -${space(1)(styleProps)}em;
      margin-right: ${space(2)(styleProps)}em;

      & {
        ${theme(styleProps.themeKey, `styles.before`)(styleProps)};
      }
    `
  };

  ${
    styleProps.isAfter &&
    css`
      margin-left: ${space(2)(styleProps)}em;
      margin-right: -${space(1)(styleProps)}em;

      & {
        ${theme(styleProps.themeKey, `styles.after`)(styleProps)};
      }
    `
  };

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const ButtonSpinnerWrapper = (styleProps) => cssClass`
  position: absolute;
  align-items: center;
  justify-content: center;

  & + .fp-Text {
    opacity: 0;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const ButtonSpinner = (styleProps) => cssClass`
  && {
    font-size: 1.25em;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const ButtonClose = (styleProps) => cssClass`
  && {
    min-height: unset;
    padding: ${space(2)(styleProps)}em;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const isInteractive = (styleProps) =>
  !styleProps.isStatic && !styleProps.isLoading && !styleProps.disabled && styleProps.variant !== 'link';

export const getDisabledProperties = (styleProps) => css`
  & {
    cursor: not-allowed;
    opacity: ${styleProps.colorMode === 'dark' ? '0.5' : '0.7'};
    outline: unset;
    pointer-events: unset;
  }
  & {
    ${theme(styleProps.themeKey, `styles.disabled`)(styleProps)};
  }
`;

export const getSizeProperties = (styleProps) => {
  const styles = {
    small: css`
      & {
        font-size: ${fontSize('100')(styleProps)}em;
        min-height: 2rem;
        padding: 0 ${space(3)(styleProps)}rem;
      }
      & {
        ${theme(styleProps.themeKey, `styles.sizes.small`)(styleProps)};
      }
    `,
    default: css`
      & {
        ${theme(styleProps.themeKey, `styles.sizes.default`)(styleProps)};
      }
    `,
    medium: css`
      & {
        min-height: 3rem;
        padding: 0 ${space(5)(styleProps)}rem;
      }
      & {
        ${theme(styleProps.themeKey, `styles.sizes.medium`)(styleProps)};
      }
    `,
    large: css`
      & {
        font-size: ${fontSize('300')(styleProps)}em;
        min-height: 3.25rem;
        padding: 0 ${space(6)(styleProps)}rem;
      }
      & {
        ${theme(styleProps.themeKey, `styles.sizes.large`)(styleProps)};
      }
    `,
  };
  return styles[styleProps.size];
};

export const getLoadingProperties = (styleProps) => css`
  & {
    cursor: not-allowed;
    position: relative;

    &:focus {
      box-shadow: unset !important;
      outline: unset !important;
    }
  }
  & {
    ${theme(styleProps.themeKey, `styles.loading`)(styleProps)};
  }
`;

export const getStaticProperties = (styleProps) => css`
  & {
    cursor: default;
    outline: unset;

    &:focus {
      box-shadow: unset !important;
      outline: unset !important;
    }
  }
  & {
    ${theme(styleProps.themeKey, `styles.static`)(styleProps)};
  }
`;

export const getInteractiveProperties = (styleProps) => css`
  &:focus {
    outline: unset;
    z-index: 2;
    box-shadow: ${palette(styleProps.palette === 'default' ? 'gray200' : styleProps.palette, {
          dark: styleProps.palette === 'default' ? 'gray700' : styleProps.palette,
        })(styleProps)}
        0px 0px 0px 1px,
      ${palette(styleProps.palette === 'default' ? 'gray100' : `${styleProps.palette}200`, {
          dark: styleProps.palette === 'default' ? 'gray600' : `${styleProps.palette}200`,
        })(styleProps)}
        0px 0px 0px 3px;

    ${styleProps.palette === 'default' &&
    css`
      border-color: transparent;
    `};

    & {
      ${theme(styleProps.themeKey, `styles.focus`)(styleProps)};
    }
  }

  ${styleProps.variant !== 'link' &&
  css`
    &:hover {
      background-color: ${palette(`${styleProps.palette === 'default' ? 'white' : styleProps.palette}600`, {
        dark: `${styleProps.palette === 'default' ? 'black100' : `${styleProps.palette}600`}`,
      })(styleProps)};

      & {
        ${theme(styleProps.themeKey, `styles.hover`)(styleProps)};
      }
    }
  `};

  ${styleProps.variant !== 'link' &&
  css`
    &:hover:active {
      background-color: ${palette(styleProps.palette === 'default' ? 'white800' : `${styleProps.palette}700`, {
        dark: `${styleProps.palette === 'default' ? 'black200' : `${styleProps.palette}700`}`,
      })(styleProps)};

      & {
        ${theme(styleProps.themeKey, `styles.hoveractive`)(styleProps)};
      }
    }
  `};
`;

export const getLinkProperties = (styleProps) => css`
  & {
    border: 0;
    background: unset;
    color: ${styleProps.palette === 'default'
      ? palette('text')(styleProps)
      : palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    fill: ${styleProps.palette === 'default'
      ? palette('text')(styleProps)
      : palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    box-shadow: unset;

    &:hover {
      text-decoration: underline;
    }
  }

  & {
    ${theme(styleProps.themeKey, `styles.link`)(styleProps)};
  }
`;

export const getOutlinedProperties = (styleProps) => css`
  & {
    background-color: ${palette('default')(styleProps)};
    border: 1px solid ${palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    color: ${palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    fill: ${palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    box-shadow: unset;

    ${isInteractive(styleProps) &&
    css`
      &:hover {
        border-color: ${palette()(styleProps)};
        background-color: ${palette(`${styleProps.palette}Tint`, { dark: `${styleProps.palette}Shade` })(styleProps)};
      }

      &:hover:active {
        border-color: ${palette()(styleProps)};
        background-color: ${palette(`${styleProps.palette}100`, { dark: `${styleProps.palette}900` })(styleProps)};
      }
    `};
  }
  & {
    ${theme(styleProps.themeKey, `styles.outlined`)(styleProps)};
  }
`;

export const getGhostProperties = (styleProps) => css`
  & {
    background-color: unset;
    border: unset;
    color: ${styleProps.palette === 'default'
      ? palette('defaultInverted')(styleProps)
      : palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    fill: ${styleProps.palette === 'default'
      ? palette('defaultInverted')(styleProps)
      : palette(styleProps.palette, { dark: `${styleProps.palette}300` })(styleProps)};
    box-shadow: unset;

    &:hover {
      background-color: rgba(0, 0, 0, 0.05);
    }
    &:hover:active {
      background-color: rgba(0, 0, 0, 0.1);
    }
  }
  & {
    ${theme(styleProps.themeKey, `styles.ghost`)(styleProps)};
  }
`;
