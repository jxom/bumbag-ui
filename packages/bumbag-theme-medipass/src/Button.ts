import { css, palette, space } from 'bumbag';
import { isInteractive } from 'bumbag/Button/styles';

export default {
  styles: {
    base: (styleProps) => css`
      border-radius: 0px;
      border: 2px solid ${styleProps.palette === 'default' ? palette('gray700')(styleProps) : 'transparent'};
      color: ${styleProps.palette === 'default'
        ? palette('gray700')(styleProps)
        : palette(`${styleProps.palette}Inverted`)(styleProps)};
      min-height: 44px;
      padding: 0 ${space(4)(styleProps)}rem;
      font-size: 15px;
      text-transform: uppercase;
    `,
    outlined: (styleProps) => css`
      background-color: ${palette('white')(styleProps)};
      border-width: 2px;

      &:hover {
        background-color: ${palette(styleProps.palette)(styleProps)};
        color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
      }

      &:hover:active {
        border-color: ${palette(`${styleProps.palette}700`)(styleProps)};
        background-color: ${palette(`${styleProps.palette}700`)(styleProps)};
      }
    `,
    ghost: (styleProps) => css`
      border: 2px solid transparent;

      &:hover {
        color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        background-color: ${palette(styleProps.palette)(styleProps)};
      }

      &:hover:active {
        color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        background-color: ${palette(styleProps.palette)(styleProps)};
      }

      &:focus {
        border: 2px solid ${palette('white')(styleProps)};
        background-color: unset;
        color: ${styleProps.palette === 'default'
          ? palette('text')(styleProps)
          : palette(styleProps.palette)(styleProps)};
      }
    `,
    focus: (styleProps) => css`
      outline: 2px solid ${styleProps.palette === 'default' ? palette('gray800')(styleProps) : palette(styleProps.palette)(styleProps)};
      outline-offset: 0;
      box-shadow: none;
      border: 2px solid ${palette('white')(styleProps)};
      background-color: ${styleProps.palette === 'default'
        ? palette('gray800')(styleProps)
        : palette(`${styleProps.palette}600`)(styleProps)};
      color: ${styleProps.palette === 'default' ? 'white' : palette(`${styleProps.palette}Inverted`)(styleProps)};
    `,
    hover: (styleProps) => css`
      ${styleProps.palette === 'default' &&
      css`
        background-color: ${palette('gray700')(styleProps)};
        color: ${palette('white')(styleProps)};
      `}
    `,
    hoveractive: (styleProps) => css`
      ${styleProps.palette === 'default' &&
      css`
        background-color: ${palette('gray800')(styleProps)};
      `}
    `,
    link: css`
      border: 2px solid transparent;
      background: unset;
    `,
    sizes: {
      small: (styleProps) => css`
        height: ${space(8)(styleProps)}rem;
        font-size: 15px;
        padding: 0 ${space(4)(styleProps)}rem;
      `,
      medium: css`
        font-size: 18px;
      `,
    },
  },
};
