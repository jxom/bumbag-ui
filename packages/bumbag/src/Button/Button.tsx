import * as React from 'react';
import { Button as ReakitButton, ButtonProps as ReakitButtonProps, useButton as useReakitButton } from 'reakit';
import ConditionalWrap from 'conditional-wrap';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { ButtonType, Omit, Size, Palette } from '../types';
import { Box, BoxProps } from '../Box';
import { Flex } from '../Flex';
import { Icon, IconProps } from '../Icon';
import { Text } from '../Text';
import { Spinner, SpinnerProps } from '../Spinner';

import * as styles from './Button.styles';

export type LocalButtonProps = {
  /** Icon that appears on the right side of the button. */
  iconAfter?: IconProps['icon'];
  iconAfterProps?: Omit<IconProps, 'icon'>;
  /** Icon that appears on the left side of the button. */
  iconBefore?: IconProps['icon'];
  iconBeforeProps?: Omit<IconProps, 'icon'>;
  /** Adds a loading indicator to the button. */
  isLoading?: boolean;
  /** Makes the button not interactable. */
  isStatic?: boolean;
  palette?: Palette;
  size?: Size;
  /** Custom props for the isLoading spinner. */
  spinnerProps?: SpinnerProps;
  type?: ButtonType;
  ignoreGrayOverride?: boolean;
};
export type ButtonProps = BoxProps & ReakitButtonProps & LocalButtonProps;

const useProps = createHook<ButtonProps>(
  (props, { themeKey }) => {
    let {
      disabled,
      focusable,
      iconAfter,
      iconAfterProps,
      iconBefore,
      iconBeforeProps,
      ignoreGrayOverride,
      spinnerProps,
      unstable_clickOnEnter,
      unstable_clickOnSpace,
      ...htmlProps
    } = props;
    if (props.palette === 'gray' && props.variant === 'default') {
      props.palette = 'grayTint';
    }

    const buttonProps = useReakitButton(
      { disabled, focusable, unstable_clickOnEnter, unstable_clickOnSpace },
      htmlProps
    );
    htmlProps = Box.useProps({ ...htmlProps, ...buttonProps });

    const className = useClassName({
      style: styles.Button,
      styleProps: props,
      themeKey,
      prevClassName: htmlProps.className,
    });
    const iconBeforeClassName = useClassName({
      style: styles.ButtonIcon,
      styleProps: { ...props, isBefore: true },
      themeKey,
      themeKeySuffix: 'Icon',
    });
    const iconAfterClassName = useClassName({
      style: styles.ButtonIcon,
      styleProps: { ...props, isAfter: true },
      themeKey,
      themeKeySuffix: 'Icon',
    });
    const spinnerWrapperClassName = useClassName({
      style: styles.ButtonSpinnerWrapper,
      styleProps: props,
      themeKey,
      themeKeySuffix: 'SpinnerWrapper',
    });
    const spinnerClassName = useClassName({
      style: styles.ButtonSpinner,
      styleProps: props,
      themeKey,
      themeKeySuffix: 'Spinner',
    });

    let spinnerColor = props.palette;
    if (props.variant === 'default') {
      spinnerColor = `${props.palette}Inverted`;
    }
    if (props.palette === 'default') {
      spinnerColor = 'defaultInverted';
    }

    let spinnerTrackColor;
    if (props.variant === 'default') {
      spinnerTrackColor = `${props.palette}300`;
    }
    if (props.palette === 'default') {
      spinnerColor = 'gray100';
    }

    const children = (
      <React.Fragment>
        {props.isLoading && (
          <Flex className={spinnerWrapperClassName}>
            <Spinner
              use={Flex}
              className={spinnerClassName}
              color={spinnerColor}
              trackColor={spinnerTrackColor}
              hasTrack
              {...spinnerProps}
            />
          </Flex>
        )}
        <ConditionalWrap condition={props.isLoading} wrap={(children) => <Text>{children}</Text>}>
          <React.Fragment>
            {iconBefore && <Icon className={iconBeforeClassName} icon={iconBefore} {...iconBeforeProps} />}
            {htmlProps.children}
            {iconAfter && <Icon className={iconAfterClassName} icon={iconAfter} {...iconAfterProps} />}
          </React.Fragment>
        </ConditionalWrap>
      </React.Fragment>
    );

    return { ...htmlProps, className, children };
  },
  {
    defaultProps: {
      disabled: false,
      iconAfter: undefined,
      iconBefore: undefined,
      isLoading: false,
      isStatic: false,
      variant: 'default',
      palette: 'default',
      size: 'default',
      type: 'button',
    },
    themeKey: 'Button',
  }
);

export const Button = createComponent<ButtonProps>(
  (props: ButtonProps) => {
    const buttonProps = useProps(props);
    return createElement({ children: props.children, component: ReakitButton, use: props.use, htmlProps: buttonProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Button',
    },
    themeKey: 'Button',
  }
);
