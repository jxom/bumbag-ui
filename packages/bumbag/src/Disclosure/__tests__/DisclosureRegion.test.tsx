import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Button } from '../../Button';
import { Blockquote } from '../../Blockquote';
import { Disclosure } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    function Component() {
      const hidden = Disclosure.useState({ baseId: 'test' });
      return (
        <div>
          <Disclosure {...hidden}>Toggle</Disclosure>
          <Disclosure.Content {...hidden} id="test">
            Hello world
          </Disclosure.Content>
        </div>
      );
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly when the hidden is visible', () => {
    function Component() {
      const hidden = Disclosure.useState({ baseId: 'test', visible: true });
      return (
        <div>
          <Disclosure {...hidden}>Toggle</Disclosure>
          <Disclosure.Content {...hidden}>Hello world</Disclosure.Content>
        </div>
      );
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('use', () => {
    it('should render correctly', () => {
      function Component() {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return (
          <div>
            <Disclosure use={Button} {...hidden}>
              Toggle
            </Disclosure>
            <Disclosure.Content use={Blockquote} {...hidden}>
              Hello world
            </Disclosure.Content>
          </div>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Disclosure props', () => {
      const { result } = renderHook(() => {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return Disclosure.Content.useProps(hidden);
      });
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      function Component() {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return (
          <Disclosure.Content {...hidden}>
            {(DisclosureProps) => <div {...DisclosureProps}>Hello world</div>}
          </Disclosure.Content>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Disclosure.base should render correctly', () => {
    const { container } = render(
      <Disclosure.Content
        baseId="test"
        overrides={{ Disclosure: { Content: { styles: { base: { backgroundColor: 'red' } } } } }}
      >
        hello world
      </Disclosure.Content>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Disclosure.base should render correctly', () => {
    const { container } = render(<Disclosure.Content baseId="test">hello world</Disclosure.Content>, {
      theme: { Disclosure: { Content: { styles: { base: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Disclosure.Content baseId="test">hello world</Disclosure.Content>, {
      theme: { Disclosure: { Content: { defaultProps: { className: 'test' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
