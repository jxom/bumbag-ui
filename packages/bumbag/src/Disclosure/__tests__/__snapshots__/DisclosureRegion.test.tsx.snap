// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Disclosure props 1`] = `
Object {
  "className": "css-1fie8gx css-63it0p css-1qnz8fc bb-Box bb-DisclosureContent",
  "hidden": true,
  "id": "test",
  "onAnimationEnd": [Function],
  "onTransitionEnd": [Function],
  "style": Object {
    "display": "none",
  },
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  Hello world
</div>
`;

exports[`composition use should render correctly 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--bb-palette-default, #000000);
  border-radius: 6px;
  color: var(--bb-palette-defaultInverted, #000000);
  fill: var(--bb-palette-defaultInverted, #000000);
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.75rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid var(--bb-palette-white900, #000000);
}

.emotion-0[disabled],
.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: var(--bb-palette-gray200, #000000) 0px 0px 0px 1px,var(--bb-palette-gray100, #000000) 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-0:hover {
  background-color: var(--bb-palette-white600, #000000);
}

.emotion-0:hover:active {
  background-color: var(--bb-palette-white800, #000000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-left: 4px solid #cccccc;
  padding: 1.5rem;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    class="emotion-0 emotion-1 emotion-2 emotion-3 bb-Box bb-Button bb-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <blockquote
    class="emotion-4 emotion-1 emotion-2 emotion-3 bb-Box bb-Blockquote bb-DisclosureContent"
    hidden=""
    id="test"
    style="display: none;"
  >
    Hello world
  </blockquote>
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;

exports[`overrides Disclosure.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;

exports[`props renders correctly when the hidden is visible 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    class="emotion-0 emotion-1 emotion-2 bb-Box bb-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-0 emotion-1 emotion-2 bb-Box bb-DisclosureContent"
    id="test"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    class="emotion-0 emotion-1 emotion-2 bb-Box bb-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-0 emotion-1 emotion-2 bb-Box bb-DisclosureContent"
    hidden=""
    id="test"
    style="display: none;"
  >
    Hello world
  </div>
</div>
`;

exports[`theming Disclosure.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;
